### PR TITLE
[OBSDEF-9031] Add KAHM preupdate healthcheck

### PR DESCRIPTION
**Pull request by tmarcu**
Signed-off-by: Tudor Marcu <Tudor.Marcu@emc.com>

- [ ] make test passed
- [ ] make build passed
- [ ] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed
- [ ] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/

## Testing
_Paste URL of charts-custom-ci job here_

_Paste the output of your helm install/vSphere7 deployment testing here_

### CODE REVIEW

**Review by vgupta-mickey — APPROVED** [review]

File: kahm/templates/kahm-app-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: decks-app-config
```
Comment (eddiepavkovic) [inline kahm/templates/kahm-app-configmap.yaml:5]:
```suggestion
  name: kahm-app-config
```

File: kahm/templates/kahm-app-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  namespace: {{.Release.Namespace}}
+  labels:
+    app.kubernetes.io/name: "decks"
```
Comment (eddiepavkovic) [inline kahm/templates/kahm-app-configmap.yaml:8]:
```suggestion
    app.kubernetes.io/name: "kahm"
```

File: kahm/templates/kahm-app-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    helm.sh/chart: {{.Chart.Name}}-{{.Chart.Version | replace "+" "_"}}
+    release: {{.Release.Name}}
+  annotations:
+    com.dellemc.kahm.subscribed: "true"
+data:
+  eventRules: |-
+  eventRemedies: |-
+  health: |-
+    spec:
+      - name: pre-update
```
Comment (eddiepavkovic) [inline kahm/templates/kahm-app-configmap.yaml:21]:
```suggestion
      - name: kahm-pre-update
```

File: kahm/values.yaml [hunk 4]
```diff
@@ -152,3 +152,11 @@ postgresql-ha:
         memory: 500Mi
       requests:
         memory: 500Mi
+
+# KAHM health check configuration
+healthChecks:
+  preUpdate:
+    image:
+      repository: kahm-pre-update
+      tag: # rfw-update-this objectstore-pre-update-image
```
Comment (eddiepavkovic) [inline kahm/values.yaml:161]:
```suggestion
      tag: # rfw-update-this kahm-pre-update-image
```

File: kahm/templates/healthchecks-rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-healthchecks
```
Comment (eddiepavkovic) [inline kahm/templates/healthchecks-rbac.yaml:5]:
```suggestion
  name: {{ .Release.Name }}-kahm-pre-update
```

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]
needs changes from decks to kahm...  would be nice to have serviceaccount as `kahm-pre-update` with limited rules.

File: kahm/templates/healthchecks-rbac.yaml (hunk 5 above)
Comment (SarahAFochs) [reply kahm/templates/healthchecks-rbac.yaml:5]: -kahm-healthchecks

**Review by youminhan — APPROVED** [review]

**Pull request merged**